server/expense.go

```diff
@@ -88,41 +110,60 @@ func listExpenses(c *gin.Context) {
 	db := GetDB()
 	expenses := []Expense{}
 
-	/* get userID from middleware */
 	userID := c.MustGet("userID")
 
 	err := db.Select(&expenses, "SELECT * FROM expense WHERE user_id=$1 ORDER BY created_at DESC", userID)
 
 	if err != nil {
-		log.Println(err)
-		c.AbortWithStatus(500)
+		saveErrorInfo(c, err, 500)
 		return
 	}
 
 	c.JSON(200, expenses)
 }
 
 func listSingleExpense(c *gin.Context) {
-	expense, expenseID := getSingleExpenseFromDB(c)
+	expenseOutput := ExpenseOutput{}
+	var err error
+	var errCode int
+	var expenseID string
+	expenseOutput.Expense, expenseID, err, errCode = getSingleExpenseFromDB(c)
 
-	if expenseID == "" {
+	if err != nil {
+		saveErrorInfo(c, err, errCode)
 		return
 	}
 
-	c.JSON(200, expense)
+	expenseOutput.Tags, err, errCode = getTagsOfExpense(c, expenseID)
+	if err != nil {
+		saveErrorInfo(c, err, errCode)
+		return
+	}
+
+	c.JSON(200, expenseOutput)
 }
 
 func updateExpense(c *gin.Context) {
-	expense, expenseID := getSingleExpenseFromDB(c)
+	expense, expenseID, err, errCode := getSingleExpenseFromDB(c)
 
-	if expenseID == "" {
+	if err != nil {
+		saveErrorInfo(c, err, errCode)
 		return
 	}
 
 	/* get updated data from body */
 	var updateExpense ExpenseInput
-	err := c.BindJSON(&updateExpense)
+	err = c.BindJSON(&updateExpense)
```

Comment:
Please always do static checks before accessing the database, so that it is not unnecessary queried